keepalived/vrrp/vrrp_scheduler.c

```diff
@@ -1094,7 +1094,7 @@ vrrp_script_child_thread(thread_t * thread)
 				log_message(LOG_INFO, "VRRP_Script(%s) timed out", vscript->sname);
 			vscript->result = 0;
 		}
-		kill(pid, SIGTERM);
+		kill(-pid, SIGTERM);
 		thread_add_child(thread->master, vrrp_script_child_timeout_thread,
```

Comment:
Yes, We do not want immediate child process terminate ahead.

This is one of the solutions. 

- Immediate child process to handle the SIGTERM.
- And send it to the PID rather than PGID.
- Immediate child process re-sent a signal to the process group
- Finaly, SIGKILL send to the process group

I also looks like there is a risk of race conditions around SIGCHLD.
Also, it is trivial, but I think also should not do a lot of things in the signal handler.
In report_child_status(), it looks like has been calls is not asynchronous-signal-safety function.

I think that it is best to run in a loop of thread_fetch.
Let me think carefully.